planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java

```diff
@@ -35,52 +37,33 @@
 
 @WebServlet("/generate-itinerary")
 public class ItineraryServlet extends HttpServlet {
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query query = new Query("Event").addSort("order", SortDirection.ASCENDING);
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(query);
+        BufferedReader br = request.getReader();
+		String eventsJson = "";
 
-        // Create an event called "hotel" with duration of 0
-        String startingAddress = request.getParameter("starting-address");
-        Event start = new Event( "Start", 
-                                startingAddress, 
-                                0);
+		if (br != null){
+			eventsJson = br.readLine();
 
-        // Get list of events from Datastore
-        List<Event> events = new ArrayList<>();
-        for (Entity entity : results.asIterable()) {
-            long id = entity.getKey().getId();
-            String name = (String) entity.getProperty("name");
-            String address = (String) entity.getProperty("address");
-            double duration = (double) entity.getProperty("duration");
-            int openingTime = Math.toIntExact((long)entity.getProperty("openingTime"));
-            int closingTime = Math.toIntExact((long)entity.getProperty("closingTime"));
-            long order = (long)entity.getProperty("order");
-            String listName = (String) entity.getProperty("listName");
-            String userId = (String) entity.getProperty("userId");
-            Event event = new Event(id, 
-                                    name, 
-                                    address, 
-                                    duration, 
-                                    TimeRange.fromStartEnd(openingTime, closingTime), 
-                                    order,
-                                    listName, 
-                                    userId);
-            events.add(event);
-        }
+            // Convert the json string to a list of events using Gson
+            Type eventListType = new TypeToken<ArrayList<Event>>(){}.getType();
+            List<Event> events = new Gson().fromJson(eventsJson, eventListType);
 
-        ItineraryGenerator itineraryGenerator = new ItineraryGenerator();
-        List<ItineraryItem> itinerary = itineraryGenerator.generateItinerary(events, start);
+            ItineraryGenerator itineraryGenerator = new ItineraryGenerator();
+            List<ItineraryItem> itinerary = itineraryGenerator.generateItinerary(events);
 
-        response.setContentType("application/json");
-        String json = new Gson().toJson(itinerary);
-        response.getWriter().println(json);
+            response.setContentType("application/json");
+            String json = new Gson().toJson(itinerary);
+            response.getWriter().println(json);
+		}else{
```

Comment:
nit: add spaces between the braces and the else: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

Reply:
Okay!